pdf/pdflinkarea.cpp

```diff
@@ -66,17 +73,36 @@ void PDFLinkArea::mousePressEvent(QMouseEvent *event)
 {
     d->clickInProgress = true;
     d->clickLocation = event->pos();
+    d->pressTimer.start();
+}
+
+void PDFLinkArea::mouseMoveEvent(QMouseEvent *event)
+{
+    // Don't activate anything if the finger has moved too far
+    QRect rect((d->clickLocation - QPointF(d->wiggleFactor, d->wiggleFactor)).toPoint(),
+               QSize(d->wiggleFactor * 2, d->wiggleFactor * 2));
+    if (!rect.contains(event->pos())) {
+        d->clickInProgress = false;
+        d->pressTimer.stop();
+        return;
+    }
 }
 
 void PDFLinkArea::mouseReleaseEvent(QMouseEvent *event)
 {
-    d->clickInProgress = false;
+    d->pressTimer.stop();
+
     // Don't activate anything if the finger has moved too far
-    QRect rect((d->clickLocation - QPointF(d->wiggleFactor, d->wiggleFactor)).toPoint(), QSize(d->wiggleFactor * 2, d->wiggleFactor * 2));
+    QRect rect((d->clickLocation - QPointF(d->wiggleFactor, d->wiggleFactor)).toPoint(),
+               QSize(d->wiggleFactor * 2, d->wiggleFactor * 2));
     if (!rect.contains(event->pos())) {
         return;
     }
 
+    event->accept();
```

Comment:
Does this affect anything? No acceptance on press done and this is also after finger move check.